[kimshitong] iP

kimshitong opened the pull request:
**KimDuke**
“Focus on being productive instead of busy.” – Tim Ferriss, American podcaster, author and entrepreneur  ([source](https://www.tameday.com/motivational-quotes-for-work-productivity/))

KimDuke frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/kimshitong/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

**Welcome to Duke**

This is my `main`
```

    public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

kimshitong committed:
commit

kimshitong committed:
commit

kimshitong committed:
Level-2 Commit

kimshitong committed:
Level 3

kimshitong committed:
Level-4

kimshitong committed:
Level-5

kimshitong committed:
Level-6

kimshitong committed:
A-TextUITEsting

kimshitong committed:
branch-Level-7 commit

kimshitong committed:
branch-Level-8 commit

kimshitong committed:
branch-A-MoreOOP commit

kimshitong committed:
Revert "branch-A-MoreOOP commit"

This reverts commit 5a8abc6257f860fe79197473b49a4afb2a76093f.

kimshitong committed:
commit moreoop

kimshitong committed:
no message

kimshitong committed:
commit

kimshitong committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

kimshitong committed:
commit gradle

kimshitong committed:
commit

kimshitong committed:
A-Jar

kimshitong committed:
commit Level-9

kimshitong committed:
A-JavaDoc commit

kimshitong committed:
commit A-CodingStandard

kimshitong committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskArray.java
#	src/main/java/duke/task/ToDo.java
#	src/main/java/duke/ui/Ui.java

kimshitong committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/TaskArray.java

kimshitong committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/TaskArray.java
#	src/main/java/duke/ui/Ui.java

seantehds commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,207 @@
+     * @return Todo Task
+     */
+    public Task processToDo() {
+        if(inputArray.length <= 1) {
```
I believe you need a whitespace here: something like
`if (inputArray.length <= 1) { `

seantehds replied:
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/94635140/addbddb4-df39-4bb3-a4db-46bfa9cf2a45)

If you wanted to see where it's from you can also go [here](https://se-education.org/guides/conventions/java/intermediate.html)

seantehds commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,207 @@
+
+    public String getExtracted() {
+
+        if(inputArray.length <= 1) {
```
Same issues as the previous comment

seantehds commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,207 @@
+            //from string to index
+            int index = Integer.parseInt(getIndex());
+            return index;
+        }catch(Exception e) {
```
Same here.

seantehds commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,207 @@
+        try {
+            int index = Integer.parseInt(getIndex());
+            return index;
+        }catch(Exception e) {
```
Yup you probably understand by this point so yeah I will stop pointing it out haha

seantehds commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,207 @@
+        boolean createdFile =createFile(this.filePath);
+
+        ArrayList<String> result = new ArrayList();
+        ArrayList<String> tobeProcessedArray = scanFile(this.filePath);
```
I understand the naming convention here, but I think maybe explicitly specifying the use of an ArrayList may be clearer? Like use `toBeProcessedArrayList` instead.

seantehds commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,207 @@
+    public boolean createFile(String filePath) {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
```
Not too sure whether this format would be allowed, as the Coding Standards state it should be done in another method instead.

![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/94635140/e6f2537b-09cd-469d-bc04-48016e4807d3)

It is right above the part on the white space issue.

seantehds commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,207 @@
+
+        ArrayList<String> lines = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
```
Same issue as above.

seantehds commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,207 @@
+     */
+    public TaskArray parseData(ArrayList<String> inputList) {
+
+        ArrayList<Task> arrayTask = new ArrayList<>();
```
Same issue as previously mentioned. Maybe `taskArrayList` may be a better name?

seantehds reviewed (commented):
Overall, I think there are just some slight issues with certain formats.

EricXiong420 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try{
```
May be missing a whitespace character. Noted the same issue in multiple other places too.

EricXiong420 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     *  Return input that has been pre-processed become sliced Array
```
Appears to have an extra whitespace character before the word "Return".

EricXiong420 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,207 @@
+     * @return inputArray
+     */
+    public String[] parseInput() {
+
```
Unnecessary line
```suggestion
```

EricXiong420 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,207 @@
+
+
+    /**
+     *  Return command after parsing the input
```
A lot of the comments have an extra whitespace character, consider removing it for standardisation.
```suggestion
     * Return command after parsing the input
```

EricXiong420 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,207 @@
+
+import duke.task.*;
+
+import java.io.*;
```
Maybe import the modules individually instead of using a wildcard import as those are not allowed and discouraged in this course?

EricXiong420 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,207 @@
+        boolean createdFolder = createFolder(this.filePath);
+        boolean createdFile =createFile(this.filePath);
+
+        ArrayList<String> result = new ArrayList();
```
This variable doesn't seem to be used in any future lines?

EricXiong420 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,64 @@
+     * @return The parsed data of the task.
+     */
+    @Override
+    public String getParsed() {
```
Maybe can say what is being parsed? eg:
```suggestion
    public String getParsedTask() {
```

EricXiong420 reviewed (commented):
Overall, good effort in following the guidelines, aside from a couple of mistakes that are prevalent throughout the code. Whitespace is missing from many of the places where they are needed (if-else, try-catch, operators, commas etc) and quite a few other coding standard violations are present too. Once you pick a few of those nits, and change some of the names to be more descriptive and fitting, the code should be good to go!

kimshitong committed:
commit

kimshitong committed:
Assertion on
1. which I assumed that there will be /from and /to in Event, also /by in deadline
2. I assumed that all taskArray in the list is being parsed and passed completely
Editing UI Display :
3. Some editting on providing better UI Display

kimshitong committed:
Commit for Code Standard
1. Naming Standard
2. spaces between if and (, switch and ( ...
3. adjusting for functions within try
4. spacing in comments

kimshitong committed:
Code Quality -
1. Avoiding Deep Nesting
2. Improved on commenting
3. Some miscilleous edits on improving code quality
4. Removing some empty spce

kimshitong committed:
Merge pull request #1 from kimshitong/branch-A-Assertions

Assertion on

kimshitong committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/kimshitong/ip

# Conflicts:
#	src/main/java/duke/parser/Parser.java

kimshitong committed:
Revert "Code Quality -"

This reverts commit aa33c01af7bf572aa0d40a51ac8430610fb5e2e5.

kimshitong committed:
re-commit... some confusion on git just now

kimshitong committed:
Reset back to assertion + origin main, before mistake merging. revert merging manually hehe, will merge again

I cant reset origin ( :( )

kimshitong committed:
Merge pull request #2 from kimshitong/branch-A-CodeQuality

Branch a code quality into merge

kimshitong committed:
no message

kimshitong committed:
commit "merge conflict"

kimshitong committed:
some follow-up for previous commit

kimshitong committed:
BCD-Extension Commit
1. Added addTag [taskIndex] [tagName]
1. Added removeTag [taskIndex]
1. Added listTag [tagName]
Some edits to improve code quality

kimshitong committed:
Merge pull request #4 from kimshitong/branch-C-Tagging

BCD-Extension Commit

kimshitong committed:
Merge pull request #3 from kimshitong/branch-A-CodeQuality

re-merge

kimshitong committed:
1. reParsed Storage
2. Improved Task's Tag Output
3. Improved UI by Varying Colour based on user with CSS
4. Revised Welcome Message
5. Modified User Photo

kimshitong committed:
1. reParsed Storage
2. Improved Task's Tag Output
3. Improved UI by Varying Colour based on user with CSS
4. Revised Welcome Message
5. Modified User Photo

kimshitong committed:
Merge pull request #6 from kimshitong/branch-A-BetterGui

Merging Branch a better gui

kimshitong committed:
Uploaded UI.png

kimshitong committed:
Revert "Uploaded UI.png"

This reverts commit f07fa2001446367d022cf7dde2c61008439ad4e7.

kimshitong committed:
uploaded ui.png

kimshitong committed:
Merge pull request #7 from kimshitong/Ui.png

merging ui.png

kimshitong committed:
Mainly updated README
Also updated some output.

kimshitong committed:
Bug Fixing on not creating data folder/file
modifying on files in test folders.

kimshitong committed:
1. Improvement on visualising text form shifting fxml from Label to Text
2. Removal of some unnecessary printing and file
